ForumProject/users/views.py

```diff
@@ -60,6 +66,76 @@ def post(self, request, *args, **kwargs):
         return response
 
 
+class RoleSelectionView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        serializer = RoleSerializer(data=request.data)
+        if serializer.is_valid():
+            user = request.user
+            if UserRoleCompany.objects.filter(user=user).exists():
+                user_role = UserRoleCompany.objects.get(user=user)
+                user_role.role = serializer.validated_data['role']
+                user_role.company_id = 0
```

Comment:
Resetting `company_id` to 0 might be problematic if 0 is not a valid `company_id`. Consider setting it to `None` if the field allows nulls to properly denote no company is selected

Reply:
Done, thanks.